dashboard/test/lib/pd/payment/payment_calculator_test.rb

```diff
@@ -33,6 +42,62 @@ class PaymentCalculatorTest < ActiveSupport::TestCase
       assert_equal 0, PaymentCalculator.instance.calculate(insufficient_puzzles)
     end
 
+    test 'Error raised if there is no payment term for workshop' do
+      error = assert_raises(RuntimeError) do
+        PaymentCalculator.instance.calculate(@csp_workshop)
+      end
+
+      assert_equal "No payment terms were found for workshop #{@csp_workshop.id}", error.message
+    end
+
+    test 'Calculate CSP Workshop with per attendee payment' do
+      create(:pd_payment_term, regional_partner: @regional_partner, properties: {per_attendee_payment: 10})
```

Comment:
nit: you don't need to specify `properties`. [SerializedProperties](https://github.com/code-dot-org/code-dot-org/blob/d3cf0d522ac5e88dd3344041a4076fb9676dd1ef/dashboard/app/models/concerns/serialized_properties.rb#L22-L24) handles the merge into properties for you. This can be written as:

```ruby
create :pd_payment_term, regional_partner: @regional_partner, per_attendee_payment: 10
```

Same below